src/routing_table/network_tests.rs

```diff
@@ -309,33 +309,39 @@ pub fn verify_network_invariant<'a, T, U>(nodes: U)
         node.verify_invariant();
     }
     // check that prefixes are disjoint
-    verify_disjoint_prefixes(groups.keys().into_iter().cloned().collect());
+    for prefix1 in groups.keys() {
+        for prefix2 in groups.keys() {
+            if prefix1 == prefix2 {
+                continue;
+            }
+            if prefix1.is_compatible(prefix2) {
+                panic!("Group prefixes should be disjoint, but these are not:\n\
+                    Group {:?}: {:?}\n\
+                    Group {:?}: {:?}",
+                       prefix1,
+                       groups[prefix1].1,
+                       prefix2,
+                       groups[prefix2].1);
```

Comment:
For the tests, it would be extremely helpful to also print the nodes whose routing tables the conflicting group information is from. Maybe `groups` should be a `HashMap<Prefix<T>, (T, HashSet<T>, HashSet<T>)>` to, in addition to its members, include the set of nodes that have that group in that form in their routing table.

The two `node.our_name` in lines 299 and 301 seem to be wrong. That could probably also be addressed if the other node's name were present in the map.